### PR TITLE
NB-4920: Porting persistent dependency installations into all custom images

### PR DESCRIPTION
**Pull request by roma-glushko**
## Summary

Setting up a user venv to persist pip installs across notebook session.

References:
- https://github.com/datarobot/notebooks/pull/2857/files

Also, I have ported a few more PRs that doesn't seem to get into custom images:
- https://github.com/datarobot/notebooks/pull/2878 (cc @Elli-Rid )

### CODE REVIEW

File: public_dropin_notebook_environments/python311_notebook/start_server.sh [hunk 1]
```diff
@@ -10,29 +11,32 @@ fi
...
 # setup the working directory for the kernel
 cd "$WORKING_DIR" || exit
 
 # setup ipython extensions
 cp -r /etc/ipython/ /home/notebooks/.ipython/
 
-# no trailing slash in the working dir path
-git config --global --add safe.directory "${WORKING_DIR%/}"
+# clear out kubernetes_specific env vars before starting kernel gateway as it will inherit them
+prefix="KUBERNETES_"; for var in $(printenv | cut -d= -f1); do [[ "$var" == "$prefix"* ]] && unset "$var"; done
```
Comment (roma-glushko) [inline public_dropin_notebook_environments/python311_notebook/start_server.sh:40]: cc @Elli-Rid

File: public_dropin_notebook_environments/README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ In this repository, we provide several example environments that you can use and
 * [Python 3.11 Notebook Environment](python311_notebook)
 * [Python 3.8 + Snowflake Notebook Environment](python38_snowflake_notebook)
 * [Python 3.9 Notebook Environment for GPU](python39_notebook_gpu)
-* [Python 3.9 Notebook Environment for GPU with Terraform](python39_notebook_gpu_tf)
+* [Python 3.9 Notebook Environment for GPU with Tensorflow](python39_notebook_gpu_tf)
```
Comment (roma-glushko) [inline public_dropin_notebook_environments/README.md:15]: @lyapun this is how I recognized you are a software engineer not a data scientist 😄

**Review by dmytrokarpovych — APPROVED** [review]
please try to build them on local before merge

File: public_dropin_notebook_environments/README.md (hunk 2 above)
Comment (lyapun) [reply public_dropin_notebook_environments/README.md:15]: lol))
sorry (

Comment (lyapun) [reply public_dropin_notebook_environments/README.md:15]: @roma-glushko could you also update it here: https://github.com/datarobot/datarobot-user-models/blob/master/public_dropin_notebook_environments/python39_notebook_gpu_tf/env_info.json#L3

I forgot to specify that it's tensorflow image in the env name

Comment (roma-glushko) [reply public_dropin_notebook_environments/README.md:15]: Yes, will do!

**Comment by roma-glushko** [comment]
@dmytrokarpovych have not built all five but just some and they look alright:
Screenshot 2024-01-03 at 14 10 23

Screenshot 2024-01-03 at 14 26 22

**Review by c-h-russell-walker — APPROVED** [review]
Looks good to me for NBX 👍

**Review by yakov-g — APPROVED** [review]

**Pull request merged**